WcaOnRails/app/models/competition.rb

```diff
@@ -458,11 +465,82 @@ def in_progress?
   end
 
   def is_over?
-    end_date < Date.today
+    !!end_date && end_date < Date.today
+  end
+
+  def results_updated_at
+    Option.from_nilable(results.order('updated_at desc').pluck(:updated_at).first)
```

Comment:
They behave the same way, but `pluck` doesn't load the entire `ActiveRecord` object and also only `SELECT`s that particular column, see http://apidock.com/rails/ActiveRecord/Calculations/pluck.